src/box/alter.cc

```diff
@@ -1748,8 +1760,13 @@ alter_space_move_indexes(struct alter_space *alter, uint32_t begin,
 		if (alter->pk_def == NULL || !index_depends_on_pk(old_index)) {
 			if (is_min_field_count_changed) {
 				new_def = index_def_dup(old_def);
-				index_def_update_optionality(new_def,
-							     min_field_count);
+				if (alter->is_space_upgrade) {
+					index_def_extend_optionality(
```

Comment:
Please drop a comment here describing when we "restore" the optionality.

Reply:
Done.